docs/design/link-sharing-service.md

```diff
@@ -0,0 +1,93 @@
+# Link Sharing Service
+
+## Abstract
+
+This design doc outlines a link sharing service that can be used by clients
+to share content with others via a simple URL.
+
+## Design
+
+### Transport
+
+The link sharing service supports HTTPS or HTTP. HTTP is intended for
+development convenience only or when offloading TLS termination to a proxy
+running on the *same host*. The link sharing service should *NOT* be deployed
+behind an off-the-box TLS terminator unless the link between the terminator and
+the link sharing service is adequately protected. See
+[Security Considerations](#security-considerations) for details.
+
+### Requests
+
+The file hosting service handles requests of the following form:
+
+`GET /<share-blob>/path/within/bucket`
+
+The `share-blob` is url-safe base64 encoding of a `Share` protobuf, which
+is defined as follows:
+
+```
+message EncryptionAccess {
+	bytes key = 1;
+
+	bytes encrypted_path_prefix = 2;
+}
+
+message Share {
+	string satellite_url = 1;
+
+	string api_key = 2;
+
+	string bucket = 3;
```

Comment:
Hmm, looking at other proto's this is defined as `bytes`? What about `api_key`?

Reply:
If I'm not 100% sure, but the `api_key` is going to be a [macaroon](https://github.com/storj/storj/blob/master/pkg/macaroon/apikey.go#L57) and the [serialize](https://github.com/storj/storj/blob/master/pkg/macaroon/apikey.go#L145) / [deserialize](https://github.com/storj/storj/blob/master/pkg/macaroon/apikey.go#L63) from a string type